### PR TITLE
refactor: group xds backend traffic features for BackendTrafficPolicy

### LINKED ISSUE #2954: Refactor ir HTTPRoute: group related features into structures
yah makes sense, something like `TrafficFeatures *TrafficFeatures` and check for `nil`

_Originally posted by @arkodg in https://github.com/envoyproxy/gateway/pull/2888#discussion_r1522007537_

### PR DESCRIPTION
**Pull request by shawnh2**
**What type of PR is this?**

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #2954, separated from #2926

### CODE REVIEW

File: internal/ir/xds.go [hunk 1]
```diff
@@ -458,6 +458,20 @@ type HTTPRoute struct {
 	Destination *RouteDestination `json:"destination,omitempty" yaml:"destination,omitempty"`
 	// Rewrite to be changed for this route.
 	URLRewrite *URLRewrite `json:"urlRewrite,omitempty" yaml:"urlRewrite,omitempty"`
+	// ExtensionRefs holds unstructured resources that were introduced by an extension and used on the HTTPRoute as extensionRef filters
+	ExtensionRefs []*UnstructuredRef `json:"extensionRefs,omitempty" yaml:"extensionRefs,omitempty"`
+	// External Processing extensions
+	ExtProcs []ExtProc `json:"extProc,omitempty" yaml:"extProc,omitempty"`
+
+	// BackendTraffic holds the features associated with BackendTrafficPolicy
+	BackendTraffic *BackendTrafficFeatures `json:"backendTraffic,omitempty" yaml:"backendTraffic,omitempty"`
```
Comment (arkodg) [inline internal/ir/xds.go:467]: prefer just `Traffic`

Comment (shawnh2) [reply internal/ir/xds.go:467]: SG

File: internal/gatewayapi/testdata/backendtrafficpolicy-status-conditions.out.yaml [hunk 2]
```diff
@@ -592,6 +592,7 @@ xdsIR:
           distinct: false
           name: ""
           prefix: /
+        traffic: {}
```
Comment (arkodg) [inline internal/gatewayapi/testdata/backendtrafficpolicy-status-conditions.out.yaml:595]: empty, is this expected ?

**Review by arkodg — APPROVED** [review]
LGTM thanks !

**Comment by zhaohuabing** [comment]
@shawnh2 resolve the conflicts, then we're good to go :-)

File: internal/gatewayapi/testdata/backendtrafficpolicy-status-conditions.out.yaml (hunk 2 above)
Comment (shawnh2) [reply internal/gatewayapi/testdata/backendtrafficpolicy-status-conditions.out.yaml:560]: indeed is empty

**Comment by shawnh2** [comment]
> @shawnh2 resolve the conflicts, then we're good to go :-)

Done

File: internal/xds/translator/route.go [hunk 3]
```diff
@@ -468,68 +479,64 @@ func buildHashPolicy(httpRoute *ir.HTTPRoute) []*routev3.RouteAction_HashPolicy
...
-			RetryOn:              retryDefaultRetryOn,
-			RetriableStatusCodes: []uint32{retryDefaultRetriableStatusCode},
-			NumRetries:           &wrapperspb.UInt32Value{Value: retryDefaultNumRetries},
-			RetryHostPredicate: []*routev3.RetryPolicy_RetryHostPredicate{
-				{
-					Name: "envoy.retry_host_predicates.previous_hosts",
-					ConfigType: &routev3.RetryPolicy_RetryHostPredicate_TypedConfig{
-						TypedConfig: protocov.ToAny(&previoushost.PreviousHostsPredicate{}),
-					},
+	rr := route.Traffic.Retry
```
Comment (shawnh2) [inline internal/xds/translator/route.go:482]: skip the nil check here, since it's been checked outside this func

**Review by arkodg — APPROVED** [review]

**Pull request merged**